vallie/vallie.go

```diff
@@ -0,0 +1,30 @@
+// Package vallie stands in for a stand-alone validation library
+// purely for discussion purposes.
+//
+// Assuming this is a desired approach, more fidelity will be added
+// and vallie would be extracted into it's own repo
+package vallie
+
+import (
+	"reflect"
+
```

Comment:
I believe the intention here is to incubate a lib that would be moved out of this repo entirely, should we go down this road.

These are orthogonal concerns, so we shouldn't ship them together if we don't have to.

The actual name "vallie" is referring to a joke Glib, Akshay, and I made on a call - I don't think any of us feel strongly as to what the actual name is called, should we determine this is the right path to head down.